test/jdk/java/io/File/GetXSpace.java

```diff
@@ -103,27 +103,50 @@ private static class Space {
         private static final long KSIZE = 1024;
         private final String name;
         private final long total;
-        private final long free;
+        private final long used;
+        private final long available;
 
-        Space(String total, String free, String name) {
+        Space(String name, String total, String used, String available) {
+            this.name = name;
             try {
                 this.total = Long.parseLong(total) * KSIZE;
-                this.free = Long.parseLong(free) * KSIZE;
+                this.used = Long.parseLong(used) * KSIZE;
+                this.available = Long.parseLong(available) * KSIZE;
             } catch (NumberFormatException x) {
                 throw new RuntimeException("the regex should have caught this", x);
             }
-            this.name = name;
         }
 
         String name() { return name; }
         long total() { return total; }
-        long free() { return free; }
+        long used() { return used; }
+        long available() { return available; }
+        long free() { return total - used; }
         boolean woomFree(long freeSpace) {
-            return ((freeSpace >= (free / 10)) && (freeSpace <= (free * 10)));
+            return ((freeSpace >= (available / 10)) &&
+                    (freeSpace <= (available * 10)));
         }
         public String toString() {
-            return String.format("%s (%d/%d)", name, free, total);
+            return String.format("%s (%d/%d/%d)", name, total, used, available);
+        }
+    }
+
+    private static void diskFree() throws IOException {
+        ArrayList<Space> al = new ArrayList<>();
+
+        String cmd = "fsutil volume diskFree C:\\";
+        StringBuilder sb = new StringBuilder();
+        Process p = Runtime.getRuntime().exec(cmd);
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
```

Comment:
The RHS could be replaced by `p.inputReader()`.  (new in JDK 17)

Reply:
So changed in 02556f265e1fc5274e73e8cfa75ce218c130112e.

@RogerRiggs Thanks for the review.